janusgraph-core/src/main/java/org/janusgraph/diskstorage/Backend.java

```diff
@@ -674,4 +668,17 @@ private static Locker openManagedLocker(String classname, String lockerName) {
             throw new IllegalArgumentException("Could not invoke method when configuring locking for: " + classname,e);
         }
     }
+
+    private ExecutorService buildExecutorService(Configuration configuration){
+
+        Integer corePoolSize = configuration.getOrDefault(PARALLEL_BACKEND_EXECUTOR_SERVICE_CORE_POOL_SIZE);
+        Integer maxPoolSize = configuration.getOrDefault(PARALLEL_BACKEND_EXECUTOR_SERVICE_MAX_POOL_SIZE);
+        Long keepAliveTime = configuration.getOrDefault(PARALLEL_BACKEND_EXECUTOR_SERVICE_KEEP_ALIVE_TIME);
+        String executorServiceClass = configuration.getOrDefault(PARALLEL_BACKEND_EXECUTOR_SERVICE_CLASS);
+
+        ExecutorServiceConfiguration executorServiceConfiguration =
+            new ExecutorServiceConfiguration(executorServiceClass, corePoolSize, maxPoolSize, keepAliveTime, null);
```

Comment:
Can we use the occasion to provide a ThreadFactory that would name those threads? Or maybe in a follow-up PR?
I.e. pass in `new ThreadFactoryBuilder().setNameFormat("backend[%02d]").build()` instead of `null`.

Reply:
I opened a separate ticket to track it: #2708 
I would prefer doing it in a follow-up PR